File: src/main/java/Models/TaskRecommendation.java
package Models;

public class TaskRecommendation {

    public enum Priority {
        HIGH, MEDIUM, LOW
    }

    public enum ActionType {
        SELECT_TASK,
        OPEN_UPDATE,
        REFRESH
    }

    private final Priority priority;
    private final String type;
    private final String message;
    private final String reason;
    private final Integer taskId;     // nullable
    private final String actionLabel;
    private final ActionType actionType;
    private final int score;

    public TaskRecommendation(Priority priority,
                              String type,
                              String message,
                              String reason,
                              Integer taskId,
                              String actionLabel,
                              ActionType actionType,
                              int score) {
        this.priority = priority;
        this.type = type;
        this.message = message;
        this.reason = reason;
        this.taskId = taskId;
        this.actionLabel = actionLabel;
        this.actionType = actionType;
        this.score = score;
    }

    public Priority getPriority() {
        return priority;
    }

    public String getType() {
        return type;
    }

    public String getMessage() {
        return message;
    }

    public String getReason() {
        return reason;
    }

    public Integer getTaskId() {
        return taskId;
    }

    public String getActionLabel() {
        return actionLabel;
    }

    public ActionType getActionType() {
        return actionType;
    }

    public int getScore() {
        return score;
    }
}

File: src/Onboarding/TaskRecommendation.php
<?php

namespace App\Onboarding;

final class TaskRecommendation
{
    public const PRIORITY_HIGH = 'high';
    public const PRIORITY_MEDIUM = 'medium';
    public const PRIORITY_LOW = 'low';

    public const ACTION_SELECT_TASK = 'select_task';
    public const ACTION_OPEN_UPDATE = 'open_update';
    public const ACTION_REFRESH = 'refresh';

    public function __construct(
        private readonly string $priority,
        private readonly string $type,
        private readonly string $message,
        private readonly string $reason,
        private readonly ?int $taskId,
        private readonly string $actionLabel,
        private readonly string $actionType,
        private readonly int $score,
    ) {
    }

    public function getPriority(): string
    {
        return $this->priority;
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function getMessage(): string
    {
        return $this->message;
    }

    public function getReason(): string
    {
        return $this->reason;
    }

    public function getTaskId(): ?int
    {
        return $this->taskId;
    }

    public function getActionLabel(): string
    {
        return $this->actionLabel;
    }

    public function getActionType(): string
    {
        return $this->actionType;
    }

    public function getScore(): int
    {
        return $this->score;
    }
}
